### PR TITLE
Bug Fix All

### PR DESCRIPTION
**Pull request by msdaungsangi**

### CODE REVIEW

File: app/Dao/CommentDao.php [hunk 1]
```diff
@@ -43,4 +43,9 @@ public function deleteCommentById(int $id)
         $comment = Comment::findOrFail($id);
         $comment->delete();
     }
+
+    public function getCommentById(int $id)
```
Comment (ingyinmaycgm) [inline app/Dao/CommentDao.php:47]: Add function doc.

File: app/Http/Controllers/Auth/ConfirmPasswordController.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?php
```
Comment (ingyinmaycgm) [inline app/Http/Controllers/Auth/ConfirmPasswordController.php:1]: For what?

File: app/Http/Controllers/Auth/ForgotPasswordController.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (ingyinmaycgm) [inline app/Http/Controllers/Auth/ForgotPasswordController.php:1]: For what?

File: app/Http/Controllers/Auth/ResetPasswordController.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (ingyinmaycgm) [inline app/Http/Controllers/Auth/ResetPasswordController.php:1]: For what?

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -90,7 +91,7 @@ public function store(Request $request)
             'password' => 'required|max:255',
             'name' => 'required|max:100',
             'img' => 'nullable|image|mimes:jpeg,png,jpg,gif|max:2024',
-            'role' => 'required',
+            'role' => 'required|in:1,2',
```
Comment (ingyinmaycgm) [inline app/Http/Controllers/UserController.php:94]: Use constant.

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -141,7 +142,7 @@ public function update(Request $request, int $id)
             'password' => 'required|max:255',
             'name' => 'required|max:100',
             'img' => 'nullable|image|mimes:jpeg,png,jpg,gif|max:255',
-            'role' => 'required',
+            'role' => 'required|in:1,2',
```
Comment (ingyinmaycgm) [inline app/Http/Controllers/UserController.php:145]: Use constant. Not hard code.

File: app/Http/Middleware/Detail.php [hunk 7]
```diff
@@ -33,7 +33,7 @@ public function handle(Request $request, Closure $next): Response
     {
         if (!Auth::check()) {
             $post = $this->postService->getPostById($request->id);
-            if ($post && $post->public_flag === Post::PUBLIC) {
+            if ($post && $post->public_flag ==='Public') {
```
Comment (ingyinmaycgm) [inline app/Http/Middleware/Detail.php:36]: Use constant.

File: app/Http/Controllers/Auth/ForgotPasswordController.php (hunk 3 above)
Comment (msdaungsangi) [reply app/Http/Controllers/Auth/ForgotPasswordController.php:1]: For forget password blade. It's a built in controller of laravel ui 4.2 .

File: app/Http/Controllers/Auth/ResetPasswordController.php (hunk 4 above)
Comment (msdaungsangi) [reply app/Http/Controllers/Auth/ResetPasswordController.php:1]: For forget password send link's blade. It's a built in controller of laravel ui 4.2 .

File: app/Http/Controllers/Auth/ConfirmPasswordController.php (hunk 2 above)
Comment (msdaungsangi) [reply app/Http/Controllers/Auth/ConfirmPasswordController.php:1]: Forget password confirmed controller. It's a built in controller of laravel ui 4.2 .

**Review by ingyinmaycgm — APPROVED** [review]

**Pull request merged**